### PR TITLE
Add some team api functions

### PR DESCRIPTION
**Pull request by jansenk**
This is for https://openedx.atlassian.net/browse/EDUCATOR-5207

* Add a function to look up team submission by student item dict
* Add a function to look up the student_ids associated with a team submission

I also added some serializer testing. My original approach for the student_ids was to add it to the serializer, and while I pivoted to an api call, I had already written a decent test for the serializer which we may as well keep.

- [x] bump version
@edx/masters-devs-gta

### CODE REVIEW

**Comment by muselesscreator** [comment]
Looks good to me, but I'm not confident enough in submission code to say its fully right.
The student_item vs student_item_dict vs student_id distinction is a bit confusing to me from the outside.

File: submissions/tests/test_team_api.py [hunk 1]
```diff
@@ -489,3 +502,62 @@ def test_reset_scores_error(self, mock_individual_reset):
...
+            individual_submission = user_to_individual_submission[user]
+            student_item_dict = individual_submission.student_item.student_item_dict
+            actual_team_submission = team_api.get_team_submission_for_student(student_item_dict)
+
+            self.assertEqual(
+                str(user_to_team_submission[user].uuid),
+                actual_team_submission['team_submission_uuid']
+            )
+
+    def test_get_team_submission_for_student__no_team_submission(self):
```
Comment (nsprenkle) [inline submissions/tests/test_team_api.py:536]: Nit: Is the double underscore in the function name intentional?

File: submissions/tests/test_team_api.py [hunk 2]
```diff
@@ -489,3 +502,62 @@ def test_reset_scores_error(self, mock_individual_reset):
...
+        self.assertEqual(
+            team_api.get_team_submission_student_ids(str(team_submission.uuid)),
+            self.student_ids
+        )
+        self.assertEqual(
+            team_api.get_team_submission_student_ids(str(team_submission_2.uuid)),
+            submission_2_student_ids
+        )
+
+    def test_get_team_submission_student_ids__no_team_submission(self):
```
Comment (nsprenkle) [inline submissions/tests/test_team_api.py:561]: Nit: Is the double underscore in the function name intentional?

File: submissions/tests/test_team_api.py (hunk 1 above)
Comment (jansenk) [reply submissions/tests/test_team_api.py:536]: yeah, it's just to separate it from the repeated prefix

File: submissions/models.py [hunk 3]
```diff
@@ -223,6 +227,31 @@ def get_team_submission_by_course_item_team(course_id, item_id, team_id):
...
+        Raises:
+            - TeamSubmissionNotFoundError if there is no matching team submission
+            - TeamSubmissionInternalError if there is some other error looking up the team submission.
+
+        """
+        try:
+            return TeamSubmission.objects.prefetch_related('submissions').get(submissions__student_item=student_item)
+        except TeamSubmission.DoesNotExist:
+            logger.error("Team submission for {} not found.".format(student_item))
+            raise TeamSubmissionNotFoundError(
```
[the remark this answers is not in the document]
Comment (jansenk) [reply submissions/models.py:244]: Everywhere else here we throw the exception, so I think it's best to remain consistent

**Pull request merged**